eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/UrlWildcardRouterPass.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * File containing the ChainRoutingPass class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * The UrlWildcardRouterPass registers UrlWildcardRouter if enabled in the configuration.
+ * It needs to be executed before \eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler\ChainRoutingPass.
+ */
+class UrlWildcardRouterPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (
+            !$container->hasParameter('ezpublish.url_wildcards.enabled') ||
+            !$container->hasParameter('ezpublish.urlwildcard_router.class')
+        ) {
+            return;
+        }
+
+        if ($container->getParameter('ezpublish.url_wildcards.enabled') === true) {
+            $definition = new Definition();
+            $definition->setClass($container->getParameter('ezpublish.urlwildcard_router.class'));
```

Comment:
I guess we can drop this given we move towards just using class name as service name, or wdyt @alongosz ?

Reply:
Definitely we should not use `.class`. It's been deprecated in Symfony 2.7 or 2.8 and dropped in 3.x (AFAIR).